.travis.yml

```diff
@@ -1,29 +1,37 @@
 language: php
+sudo: false
 
 php:
   - 5.6
+  - 7.0
+  - 7.1
 
 env:
-  - MONGO_DRIVER=mongo
+  global:
+    - DRIVER_VERSION="stable"
+    - ADAPTER_VERSION="^1.0.0"
 
 matrix:
   include:
     - php: 5.6
-      env: MONGO_DRIVER=mongo COMPOSER_FLAGS='--prefer-lowest --prefer-stable'
-    - php: 7.0
-      env: ADAPTER_VERSION="^1.0.0" MONGO_DRIVER=mongodb
-
-sudo: false
+      env: DRIVER_VERSION="1.5.8" PREFER_LOWEST="--prefer-lowest"
 
 cache:
   directories:
     - $HOME/.composer/cache
 
 services: mongodb
 
-before_install:
-  - yes '' | pecl -q install -f $MONGO_DRIVER
-  - if [ "x${ADAPTER_VERSION}" != "x" ]; then composer require "alcaeus/mongo-php-adapter=${ADAPTER_VERSION}" --ignore-platform-reqs; fi
-
 install:
-  - composer update $COMPOSER_FLAGS
+  - composer self-update
+  - if [[ ${TRAVIS_PHP_VERSION:0:2} == "5." ]]; then yes '' | pecl -q install -f mongo-${DRIVER_VERSION}; fi
+  - if [[ ${TRAVIS_PHP_VERSION:0:2} == "7." ]]; then pecl install -f mongodb-${DRIVER_VERSION}; fi
+  - if [[ ${TRAVIS_PHP_VERSION:0:2} == "7." ]]; then composer require "alcaeus/mongo-php-adapter=${ADAPTER_VERSION}" --ignore-platform-reqs; fi
+  - composer update ${PREFER_LOWEST}
```

Comment:
you should add ``--prefer-dist``, which is better for caching (now that you use the container-based system which has access to it)